pkg/netpol/diff/diff.go

```diff
@@ -107,6 +107,8 @@ func (da *DiffAnalyzer) ConnDiffFromDirPaths(dirPath1, dirPath2 string) (Connect
 		da.errors = append(da.errors, newConnectionsAnalyzingError(err))
 		return nil, err
 	}
+	// peers names found in the dirs (some peers may not appear in the conns, so we need to list all peers in the resources)
+	workloadsNames1, workloadsNames2 := caAnalyzer.GetPeersNamesFromDirPath(dirPath1), caAnalyzer.GetPeersNamesFromDirPath(dirPath2)
```

Comment:
can  avoid the changes in `connlist` and just compute the workloads here, on the result returned from `ConnlistFromDirPath`. The function `getIPblocksFromConnList` does something similar - it retrieves the ip-blocks from the connlist result. You can add another function to retrieve the workloads from that result.